iRecruit-ui/src/components/Profile/Profile.jsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useEffect } from "react";
+import './Profile.css';
+
+function Profile() {
+  const [photo, setPhoto] = useState(null);
+  const [bio, setBio] = useState("");
+  const [education, setEducation] = useState("");
+  const [currentInstitution, setCurrentInstitution] = useState("");
+  const [skills, setSkills] = useState("");
+  const [experience, setExperience] = useState("");
+  const [linkedinUrl, setLinkedinUrl] = useState("");
+  //all of these can be updated, hence the need for state hooks
+  const [profileUpdated, setProfileUpdated] = useState(false);
+
+  useEffect(() => {
+    // Retrieve the profile information from localStorage when the component mounts
+    const storedProfile = localStorage.getItem("profile");
+    if (storedProfile) {
+      const { photo, bio, education, currentInstitution, skills, experience, linkedinUrl } = JSON.parse(storedProfile);
+      setPhoto(photo);
+      setBio(bio);
+      setEducation(education);
+      setCurrentInstitution(currentInstitution);
+      setSkills(skills);
+      setExperience(experience);
+      setLinkedinUrl(linkedinUrl);
+    }
+  }, []);
+
+  const handlePhotoChange = (event) => {
+    const file = event.target.files[0];
+    if (file) {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        setPhoto({
+          file,
+          dataURL: reader.result,
+        });
+      };
+      reader.readAsDataURL(file);
+    } else {
+      setPhoto(null);
+    }
+  };
+
+  const handleBioChange = (event) => {
+    setBio(event.target.value);
+  };
+
+  const handleEducationChange = (event) => {
+    setEducation(event.target.value);
+  };
+
+  const handleCurrentInstitutionChange = (event) => {
+    setCurrentInstitution(event.target.value);
+  };
+
+  const handleSkillsChange = (event) => {
+    setSkills(event.target.value);
+  };
+
+  const handleExperienceChange = (event) => {
+    setExperience(event.target.value);
+  };
+
+  const handleLinkedinUrlChange = (event) => {
+    setLinkedinUrl(event.target.value);
+  };
```

Comment:
This is a lot of duplicate code, it'd be better to do something like:
```
const handleUpdate = (updateType, event) => {
  const { target: { value } } = event;
  switch(updateType) {
    case UPDATE_TYPES.BIO:
       setBio(value);
     ...
  }
}

```

Reply:
Okay, I will see to it, I am not very familiar with this pattern yet but I will look it up and see how  I can best implement it.